comms.js

```diff
@@ -37,8 +37,8 @@ Comms.prototype.write = function(msg) {
 	}
 };
 
-Comms.prototype.listen = function(port) {
-	var self = this, server = this.server = jot.createServer(port);
+Comms.prototype.listen = function(port, opts) {
+	var self = this, server = this.server = jot.createServer(opts || {}, port);
```

Comment:
Ah, the wonderful world of maintaining API compatibility...

The `port` argument to `Comms#listen` is quietly overloaded.  It can accept *either* a numeric port number or an object containing options for `createServer` (L41) **and** a `port` number for `server.listen()` (L50).

This worked because node [used to be much looser about the type of the argument passed in to `createServer`](https://github.com/nodejs/node/blob/v4.6.2/lib/net.js#L1098-L1108).  If it wasn't a function, it was assumed the argument was an object.  This was OK because primitive value types (like numeric values) **do** have property accessors that return `undefined`.  eg:

```js
options = 1234;
options.foo // => undefined
```
...which means that where node looks for options, everything works as if you had passed in `{}`.  (node 6 broke this behavior by [being stricter about the type of arguments accepted by `createServer`](https://github.com/nodejs/node/blob/v6.9.1/lib/net.js#L1115-L1117).)

Then on L50, the `listen()` method accepts either [a numeric port](https://nodejs.org/dist/latest-v6.x/docs/api/net.html#net_server_listen_port_hostname_backlog_callback) or [an options object](https://nodejs.org/dist/latest-v6.x/docs/api/net.html#net_server_listen_options_callback).  If it's an object with a `port` property, [`listen()` happily uses that](https://github.com/nodejs/node/blob/v6.9.1/lib/net.js#L1349-L1371).

So now we're in the situation where `Comms#listen` exposes an overloaded API that accepts either a numeric port number **or** a options object that's actually combined to provide options to two separate APIs (`createServer` and `listen`).  This works because those two APIs use differently-named properties for their options (which is perhaps a bit too clever, but that's not something we can fix in a semver-minor patch).

In order for this to not be a breaking change, we need to continue supporting the old behavior: if `port` is a number, call `jot.createServer()` with no arguments.  If it is an object, pass in the object.  (The `listen` call on L50 stays untouched.)